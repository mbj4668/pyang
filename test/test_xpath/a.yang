module a {
  yang-version 1.1;
  namespace "urn:a";
  prefix a;

  include as;
  import b {
    prefix b;
  }

  container x {
    uses b:foo {
      when "foo"; // FIXME: should be error, 'foo' is in the grouping!
    }
    action doit {
      input {
        must "../doit/now or ../bar";
        container now {
          when "../../bar";
        }
      }
    }
    container bar {
      leaf q {
        type string;
      }
      must "position()/foo"; // error, bad func ret
      must "current()/foox"; // error, foox not defined
      must "q[current()/q]"; // ok
    }
    container y {
      when '/bar'; // error, /bar doesn't exist
    }
    choice c1 {
      case c2 {
        container c {
          when '../d';
        }
        container d {
          when '../e'; // FIXME: should be error, e defined in other case!
        }
      }
      container e {
        when '../y'; // ok to refer to node outside the choice
      }
      case c3 {
        choice c4 {
          when 'y'; // ok to refer to node outside the case
          case c5;
        }
      }
    }
  }

  grouping foo {
    leaf a {
      type int32;
      must "current( = 'xpdr'"; // syntax error
      must "foo"; // syntax ok
      when "current( = 'xpdr'"; // syntax error
    }
  }
  uses foo; // ensure we don't get xpath errors again, but a warning for 'foo'

}
