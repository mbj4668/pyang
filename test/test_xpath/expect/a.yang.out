a.yang:27: error: XPATH_FUNCTION_RET_VAL
a.yang:28: warning: XPATH_NODE_NOT_FOUND1
a.yang:32: warning: XPATH_NODE_NOT_FOUND2
a.yang:58: error: XPATH_SYNTAX_ERROR
a.yang:63 (at a.yang:59): warning: XPATH_NODE_NOT_FOUND1
a.yang:60: error: XPATH_SYNTAX_ERROR
as.yang:9 (at b.yang:7): warning: XPATH_NODE_NOT_FOUND2
